src/components/ContentGroups/CampaignLoanDisplay.vue

```diff
@@ -0,0 +1,236 @@
+}
+
+export default {
+	inject: ['apollo'],
+	components: {
+		// KvIcon,
+		KvLoadingOverlay,
+		KvPagination,
+		LoanCardController,
+	},
+	props: {
+		checkoutVisible: {
+			type: Boolean,
+			default: false,
+		},
+		filters: {
+			type: Object,
+			default: () => {},
+		},
+		isVisitor: {
+			type: Boolean,
+			default: true,
+		},
+		itemsInBasket: {
+			type: Array,
+			default: () => [],
+		}
+	},
+	data() {
+		return {
+			limit: loansPerPage,
+			loadingLoans: false,
+			loans: [],
+			loansPerPage,
+			offset: 0,
+			pageQuery: { page: '1' },
+			totalCount: 0,
+		};
+	},
+	computed: {
+		urlParams() {
+			return toUrlParams({
+				offset: this.offset,
+			});
+		},
+		lastLoanPage() {
+			return Math.ceil(this.totalCount / this.limit);
+		},
+		loanQueryVars() {
+			return {
+				limit: this.limit,
+				loans: () => [],
+				offset: this.offset,
+				filters: this.filters,
+				promoOnly: { basketId: cookieStore.get('kvbskt') }
+			};
+		},
+	},
+	created() {
+		// extract query
+		this.pageQuery = this.$route.query;
+	},
+	methods: {
+		addToBasket(payload) {
+			this.$emit('add-to-basket', payload);
+		},
+
+		activateLoanWatchQuery() {
+			const observer = this.apollo.watchQuery({
+				query: basicLoanQuery,
+				variables: this.loanQueryVars
+			});
+			this.$watch(() => this.loanQueryVars, vars => {
+				observer.setVariables(vars);
+			}, { deep: true });
+			// Subscribe to the observer to see each result
+			observer.subscribe({
+				next: ({ data, loading }) => {
+					if (loading) {
+						this.loadingLoans = true;
+					} else {
+						this.loans = data.lend?.loans?.values ?? [];
+						this.totalCount = data.lend?.loans?.totalCount ?? 0;
+						this.checkIfPageIsOutOfRange(this.loans.length, this.pageQuery.page);
+						this.loadingLoans = false;
+					}
+				}
+			});
+		},
+
+		// Pagination Related methods
+		checkIfPageIsOutOfRange(loansArrayLength, pageQueryParam) {
+			// determines if the page query param is for a page that is out of bounds.
+			// if it is, changes page to the last page and displays a tip message
+			const loansOutOfRange = loansArrayLength === 0 && pageQueryParam;
+			if (loansOutOfRange) {
+				// eslint-disable-next-line max-len
+				this.$showTipMsg(`There are currently ${this.lastLoanPage} pages of results. We’ve loaded the last page for you.`);
+				this.pageChange(this.lastLoanPage);
+			}
+		},
+		pageChange(number) {
+			const offset = loansPerPage * (number - 1);
+			this.offset = offset;
+			this.pushChangesToUrl();
+		},
+		updateFromParams(query) {
+			const { offset } = fromUrlParams(query);
+			this.offset = offset;
+		},
+		pushChangesToUrl() {
+			const { page } = this.$route?.query ?? { page: '0' };
+			if (page !== this.urlParams.page) {
+				this.$router.push({
+					query: {
+						...this.$route.query,
+						...this.urlParams
+					}
+				});
+			}
+		},
+	},
+	// TODO: determine if this logic will work at this level
```

Comment:
It won't work since those guard are only available to the components passed to the router in routes.js. You can possibly do all query/urlToQuery logic and transformations in CCLanding page and pass it to this component as a prop?

Reply:
Yea that's what I thought I remembered too. The route logic only works on the top level component served to the route.

I'm still working out the isolation of pagination logic which this is related to. I can probably do what you've suggested once I create a utils file for pagination, which uses some additional functions to format the `offset`. This is no concern of the page level component so I'd rather not require it at this level. _It's still defined in the data object here but is not used, still a bit more clean up here..._